lib/LaTeXML/Core/Gullet.pm

```diff
@@ -254,15 +255,13 @@ sub readXToken {
     if (!defined $token) {
       return unless $$self{autoclose} && $toplevel && @{ $$self{mouthstack} };
       $self->closeMouth; }    # Next input stream.
-    elsif (my $unexpanded = $token->is_notexpanded) {
-      # Should only occur IMMEDIATELY after expanding \noexpand (by readXToken),
-      # so this token should never leak out through an EXTERNAL call to readToken.
-      return $unexpanded; }    # Just return the next token.
     elsif (($cc = $$token[1]) == CC_COMMENT) {    # NOTE: Inlined ->getCatcode
       return $token if $commentsok;
       push(@{ $$self{pending_comments} }, $token); }    # What to do with comments???
     elsif ($cc == CC_MARKER) {
       LaTeXML::Core::Definition::stopProfiling($token, 'expand'); }
+    elsif ($$token[2]) {                                # Inline is_dont_expand
+      return $$token[2]; }
     # Note: special-purpose lookup in State, for efficiency
```

Comment:
since we're grasping for efficiency, isn't ```my $unexpanded = $$token[2]``` infinitesimally better?

Reply:
indeed